lib/components/user/with-logged-in-user-support.js

```diff
@@ -37,33 +37,34 @@ export default function withLoggedInUserSupport (WrappedComponent, requireLogged
  * Upon completion (or if no user is logged in or if auth is disabled), it renders children.
  */
 class UserLoaderScreen extends Component {
+  /**
+   * Determines whether user data should be fetched.
+   * @returns true if the logged-in user has passed Auth0 authentication and state.user.loggedInUser has not been set; false otherwise.
+   */
+  shouldFetchUser = () => {
+    const { auth0, loggedInUser } = this.props
+    return auth0 && auth0.isAuthenticated && !loggedInUser
+  }
+
   async componentDidUpdate () {
-    const { auth0, fetchOrInitializeUser, loggedInUser } = this.props
+    const { auth0, fetchOrInitializeUser } = this.props
 
-    if (auth0 && auth0.isAuthenticated && !loggedInUser) {
-      try {
-        // Get the token
-        const token = await auth0.getAccessTokenSilently()
-        // Once accessToken is available, proceed to fetch or initialize loggedInUser.
-        fetchOrInitializeUser(auth0, token)
-      } catch (error) {
-        alert('Error obtaining an authorization token.')
-      }
+    if (this.shouldFetchUser()) {
+      fetchOrInitializeUser(auth0)
     }
   }
 
   render () {
-    const { auth0, children, loggedInUser, requireLoggedInUser } = this.props
+    const { auth0, children, requireLoggedInUser } = this.props
 
-    if (auth0) {
-      if (requireLoggedInUser && auth0.isAuthenticated && !loggedInUser) {
-        // Display a hint while fetching user data for logged in user (from componentDidMount).
-        // Don't display this if loggedInUser is already available.
-        // TODO: Improve this screen.
-        return <AwaitingScreen />
-      } else {
-        return renderChildrenWithProps(children, { auth0 })
-      }
+    if (requireLoggedInUser && this.shouldFetchUser()) {
+      // If a logged-in user is required, then
+      // display a hint while the logged-in user data is being fetched (from componentDidMount).
+      // Don't display this if loggedInUser is not required or is already available.
+      // TODO: Improve this screen.
+      return <AwaitingScreen />
+    } else if (auth0) {
+      return renderChildrenWithProps(children, { auth0 })
```

Comment:
"else if" condition could use a comment.

Reply:
Glad you pointed this out. I ended up removing the else-if block as only one component was using the `auth0` prop to get the `email_verified` state. I added an `isEmailVerified` to the user state in https://github.com/opentripplanner/otp-react-redux/pull/233/commits/8fe223ef4bda834432f6ecc59d3162180af76650, I think that looks better.

Reply:
Hmm, after taking a closer look at https://github.com/opentripplanner/otp-react-redux/commit/8fe223ef4bda834432f6ecc59d3162180af76650, it feels a little awkward to me. Here's a few suggestions:
- add default state for create-user-reducer. It can be confusing when there is no initial state.
- in `UserLoaderScreen#componentDidUpdate`:
  - remove `async`
  - add `prevProps` arg and compare prevProps.auth0.email_verified to this.props.... etc.
  - if this.props.email verified is true, dispatch new action (SET_EMAIL_VERIFIED) to true.
  - move email verified logic in user reducer to new case

Reply:
Should the Auth0 state be stored in the redux state?

Reply:
I don't think we should expect the email verified state to change, so there shouldn't be an action for that (unless when the user clicks "My email is verified", we do something other than reloading the page).

Reply:
Yea, perhaps storing the Auth0 state alongside the loggedInUser would be a better approach. Do you see any issues there?

Reply:
Adding the Auth0 state in redux would help with another refactor I have been considering doing on the login button.

Reply:
The `isEmailVerified` hack is removed in https://github.com/opentripplanner/otp-react-redux/pull/233/commits/e25ef5d781232cde60c0ac866b357b7610e65d47.

Reply:
I reversed course and went back to passing the `auth0` prop to `UserAccountScreen` (found a better way to do that in https://github.com/opentripplanner/otp-react-redux/pull/233/commits/d359a8b8930d10602543081ec3d7e16511180dae). I don't think storing `auth0` in the redux state is warranted yet.

Reply:
Nice work!